examples/nivisa_dmm_measurement/_visa_dmm.py

```diff
@@ -0,0 +1,136 @@
+"""Custom instrument driver for MeasurementLink NI-VISA DMM examples."""
+from __future__ import annotations
+
+import pathlib
+import sys
+from enum import Enum
+from types import TracebackType
+from typing import (
+    TYPE_CHECKING,
+    Optional,
+    Type,
+)
+
+import pyvisa
+import pyvisa.resources
+import pyvisa.typing
+
+
+if TYPE_CHECKING:
+    if sys.version_info >= (3, 11):
+        from typing import Self
+    else:
+        from typing_extensions import Self
+
+
+# Pin map instrument type constants for VISA DMM Simulator
+INSTRUMENT_TYPE_VISA_DMM = "VisaDmm"
+
+_SIMULATION_YAML_PATH = pathlib.Path(__file__).resolve().parent / "_visa_dmm_sim.yaml"
+
+_RESOLUTION_DIGITS_TO_VALUE = {"3.5": 0.001, "4.5": 0.0001, "5.5": 1e-5, "6.5": 1e-6}
+
+# Supported NI-VISA DMM instrument IDs, both real and simulated, can be added here
+_VALID_INSTRUMENT_ID = [INSTRUMENT_TYPE_VISA_DMM, "34401"]
```

Comment:
`INSTRUMENT_TYPE_VISA_DMM` is only for the pinmap's `Instrument/@instrumentTypeId` attribute. It is not an instrument ID that would be returned from `*IDN?`. It normally would not belong in an instrument driver, because instrument drivers don't know anything about pin maps.

Replace it with "Waveform Generator Simulator" so that this example continues to work with a real NI Instrument Simulator v2.0.